lockup/README.md

```diff
@@ -1,107 +1,110 @@
 
-### Release schedule
+### Combining lockup and vesting
 
-The release of tokens can also be scheduled to be linear from the moment tokens are unlocked.
-The tokens are unlocked at the following timestamp `max(transfers_enabled_timestamp + lockup_duration, lockup_timestamp)`.
+The contract could have both lockup and vesting schedules.
+
+The tokens are started to unlock at the following timestamp `max(max(transfers_enabled_timestamp + lockup_duration, lockup_timestamp), cliff_timestamp)`.
 NOTE: If the `lockup_timestamp` is not specified, the tokens are unlocked at `transfers_enabled_timestamp + lockup_duration`.
 
-Release schedule can not be terminated by the foundation.
+The current amount of locked tokens are calculated as the maximum between lockup and vesting logic.
+If at least one mechanism said the tokens are locked, then they are still locked.
+
+The contract could also have only one of these mechanisms. When initializing, it's possible to pass empty vesting information.
+It's also possible not to provide `release_duration`, it means that we are using a vesting schedule.
+If neither of the mechanisms is initialized, the tokens will be locked forever.
 
 ### Staking
 
-NEAR is the proof of stake network. The owner of the lockup contract might hold large percentage of the network tokens.
-The owner may want to stake these tokens (including locked tokens) to help secure the network and also earn staking rewards that are distributed to network validator.
+NEAR is the proof of stake network. The owner of the lockup contract might hold a large percentage of the network tokens.
+The owner may want to stake these tokens (including locked tokens) to help secure the network and also earn staking rewards that are distributed to the network validator.
 This contract doesn't allow to directly stake from this account, so the owner can delegate tokens to a staking pool contract (see https://github.com/near/initial-contracts/tree/master/staking-pool).
 
-The owner can choose the staking pool where to delegate tokens.
-The staking pool contract and the account has to be approved and whitelisted by the foundation, to prevent lockup tokens from being lost, locked or stolen.
+The owner can choose the staking pool for delegating tokens.
+The staking pool contract and the account have to be approved and whitelisted by the NEAR Foundation, to prevent lockup tokens from being lost, locked, or stolen.
 This staking pool must be an approved account, which is validated by a whitelisting contract.
-Once the staking pool holds tokens, the owner of the staking pool is able to use them to vote on the network governance issues, such as enabling transfers.
-So it's important for the owner to pick the staking pool that fits the best.
+Once the staking pool holds tokens, the owner of the staking pool can use them to vote on the network governance issues, such as enabling transfers.
```

Comment:
Is it hardcoded in the contract? Could we please link the place from the source code in the documentation?

Reply:
I think this part is obsolete and should be deprecated. Since the voting for transfers has already passed there is no reason for the new lockup contracts to be initialized without explicitly enabled transfers with the fixed timestamp.

The account_id that was checking and counting votes is not hardcoded, but a part of the enum:
https://github.com/near/core-contracts/blob/d84f7a924b79009d3f7aa4e577c07b08a0489dc9/lockup/src/types.rs#L62

I think we shouldn't use this anymore and should explicitly mark it as deprecated in the code.

Reply:
I created a separate issue about marking as deprecated/fixing comment in the contract: https://github.com/near/core-contracts/issues/146
Please comment there anything that you remember that should be fixed